kit/inputs/src/UIAmountInput/hooks/useSetText.ts

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import BigNumber from 'bignumber.js';
+import { runOnJS, useAnimatedRef, useWorkletCallback } from 'react-native-reanimated';
+import type { UITextViewRef } from '../../UITextView/types';
+import { AmountInputContext } from '../constants';
+import { setTextAndCaretPosition } from '../setTextAndCaretPosition';
+import type { FormatAndSetTextConfig, SetText, TextAttributes, UIAmountInputProps } from '../types';
+
+const defaultConfig: FormatAndSetTextConfig = {
+    shouldSetText: true,
+    callOnChangeProp: true,
+};
+export function useSetText(
+    ref: React.RefObject<UITextViewRef>,
+    onChangeAmountProp: UIAmountInputProps['onChangeAmount'],
+): SetText {
+    // @ts-expect-error
```

Comment:
Also, what's the problem?

Reply:
Also problem with the types don't match with `useAnimatedRef`